recipes/type_safe/meta.yaml

```diff
@@ -0,0 +1,38 @@
+{% set name = "type_safe" %}
+{% set version = "0.2.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/foonathan/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 34d97123fb9bca04a333565c4a2498425d602ec0c759de4be1b8cfae77d05823
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - cmake
+    # type_safe is header-only but its cmake setup is looking for a compiler
+    - {{ compiler('cxx') }}
```

Comment:
Could you please try disabling building the examples/tests using this option, and remove this compiler requirement?

https://github.com/foonathan/type_safe/blob/main/CMakeLists.txt#L143

Otherwise, this package will have implied runtime dependencies that need to be explicitly ignored.

I assume you want some cmake artifacts to be installed along with the headers otherwise, you could just copy the headers directly instead of using cmake.

Reply:
Thanks for the feedback. Actually, this did not work. cmake will look for a C++ compiler even if we never compile anything. We'd have to set `project(... LANGUAGES NONE)` to remove the compiler check. But then we probably also need to patch other parts of the setup.

I am now ignoring the constraints added by the C++ compiler.